alternator/executor.cc

```diff
@@ -3071,19 +3071,14 @@ future<> executor::cas_write(schema_ptr schema, service::cas_shard cas_shard, dh
                                     trace_state = tracing::trace_state_ptr(gt), &self]
                                     (service::client_state& client_state) mutable {
                 auto schema = self._proxy.data_dictionary().find_schema(ks, cf);
-
-                // The desired_shard on the original shard remains alive for the duration
-                // of cas_write on this shard and prevents any tablet operations.
-                // However, we need a local instance of cas_shard on this shard
-                // to pass it to sp::cas, so we just create a new one.
                 service::cas_shard cas_shard(*schema, dk.token());
```

Comment:
I don't understand this change. What do you mean "We no longer depend on keeping the cas_shard on the original shard for correctness" - did you change this in this PR or it's something that was generally changed in Scylla?
You still need to leave **some** comment to explain why we need this bizarre cas_shard variable, perhaps explaining that we don't need it for locking erm but still needs to supply a new one to pass to sp::cas (but please explain it better...).

Perhaps to reduce confusion (you also have a cas_shard variable in the outer scope) you can rename this variable? The original code did not have this confusion because one variable was called desired_shard and the other cas_shard.